ENIGMAsystem/SHELL/Graphics_Systems/Direct3D9/DX9vertex.cpp

```diff
@@ -15,145 +15,131 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
-#include <vector>
+#include "Graphics_Systems/General/GSvertex_impl.h"
+#include "Graphics_Systems/General/GSprimitives.h" // for enigma_user::draw_primitive_count
+#include "Graphics_Systems/General/GScolor_macros.h"
+
+#include "Bridges/General/DX9Context.h"
+
 #include <map>
-using std::vector;
 using std::map;
 
-#include "../General/GSvertex.h"
-
 namespace enigma {
 
-struct VertexBuffer {
-	vector<gs_scalar> vertices;
-	vector<gs_scalar> indices;
-	
-};
-
-vector<VertexFormat*> vertexFormats;
-vector<VertexBuffer*> vertexBuffers;
-	
-VertexFormat* vertexFormat = 0;
-
-}
-
-namespace enigma_user {
-
-int vertex_create_buffer() {
-
-}
-
-int vertex_create_buffer_ext(unsigned size) {
-
-}
-
-void vertex_delete_buffer(int buffer) {
-
-}
-
-void vertex_begin(int buffer, int format) {
-
-}
-
-void vertex_end(int buffer) {
-
-}
-
-void vertex_freeze(int buffer) {
-
-}
-
-void vertex_submit(int buffer, int primitive) {
-
-}
-
-void vertex_submit(int buffer, int primitive, int texture) {
-
-}
-
-void vertex_delete(int buffer) {
-
-}
+D3DPRIMITIVETYPE primitive_types[] = { static_cast<D3DPRIMITIVETYPE>(0), D3DPT_POINTLIST, D3DPT_LINELIST, D3DPT_LINESTRIP, D3DPT_TRIANGLELIST, D3DPT_TRIANGLESTRIP, D3DPT_TRIANGLEFAN };
+D3DDECLTYPE declaration_types[] = { D3DDECLTYPE_FLOAT1, D3DDECLTYPE_FLOAT2, D3DDECLTYPE_FLOAT3, D3DDECLTYPE_FLOAT4, D3DDECLTYPE_D3DCOLOR, D3DDECLTYPE_UBYTE4 };
+WORD declaration_type_sizes[] = { 4, 8, 12, 16, 4, 4 };
```

Comment:
Why not use `sizeof`? It's safer than hard-coding it. Also safe: use fixed array bounds.

Reply:
`D3DDECLTYPE_FLOAT1` is an enum value, I could not find a macro to obtain these sizes and every example online simply hardcodes this.

Reply:
Yeah, after looking some more, and through all the D3D macros/constants/functions I couldn't find anything to give me the size. So I did the safer thing and used sizeof like you requested: 5c43677c6fb2b993a4a318fcfff7cb265fbb9816